libs/vkd3d/state.c

```diff
@@ -1874,6 +1880,17 @@ static ULONG STDMETHODCALLTYPE d3d12_pipeline_state_Release(ID3D12PipelineState
 
         d3d12_device_release(device);
```

Comment:
Do we test the device ref count behaviour? Normally we're supposed to release the device when the **public** ref count reaches 0, and AddRef the device when the public ref count goes from 0 to 1.

Probably not the end of the world here since the pipeline objects are owned by the pipeline library rather than the device itself, but generally this is done to avoid circular references.

Reply:
I can test this.

Reply:
Correct, we have to release the device reference, but not the private data. Just d3d12 memes ...